api/src/Tests/Cases/OAuth2Test.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace App\Tests\TestCase\Controller;
+
+use App\Tests\Base\CiabTestCase;
+
+class OAuth2Test extends CiabTestCase
+{
+
+
+    protected function setUp(): void
+    {
+        $this->setupToken = false;
+        parent::setUp();
+
+    }
+
+
+    public function testTokens(): void
+    {
+        $this->runRequest('POST', '/token', null, ['grant_type' => 'passwordx', 'username' => self::$login, 'password' => self::$password, 'client_id' => self::$client], 400);
```

Comment:
One trick: look at PHPUnit's docco into data providers to DRY this up a little. It's nifty.

Reply:
Provider figured out and used